packages/components/pages/f-contact-preferences/src/store/_tests/contactPreferences.module.test.js

```diff
@@ -0,0 +1,149 @@
+
+                // Act
+                contactPreferencesModule.actions.editPreference({ state: contactPreferencesViewModel, commit: commitSpy }, { key: 'news', field: 'emailValue', value: expectedValue });
+
+                // Assert
+                expect(commitSpy).toHaveBeenLastCalledWith(UPDATE_PREFERENCE, { index: expectedIndex, field: 'emailValue', value: expectedValue });
+            });
+
+            it(`should not call ${UPDATE_PREFERENCE} mutation if the key not found`, () => {
+                // Arrange
+                const commitSpy = jest.fn();
+
+                // Act
+                contactPreferencesModule.actions.editPreference({ state: contactPreferencesViewModel, commit: commitSpy }, { key: 'X-X-X', field: 'emailValue', value: true });
+
+                // Assert
+                expect(commitSpy).not.toHaveBeenCalled();
+            });
+
+            it(`should not call ${UPDATE_PREFERENCE} mutation if the field not found`, () => {
+                // Arrange
+                const commitSpy = jest.fn();
+
+                // Act
+                contactPreferencesModule.actions.editPreference({ state: contactPreferencesViewModel, commit: commitSpy }, { key: 'news', field: 'X-X-X', value: true });
+
+                // Assert
+                expect(commitSpy).not.toHaveBeenCalled();
+            });
+        });
+    });
+
+    describe('mutations ::', () => {
+        describe(`${UPDATE_PREFERENCES} ::`, () => {
+            it('should update preference state with mapped/filtered/sorted preference data', () => {
+                // Arrange
+                const expectedPreferenceVersionViewed = 1;
+                const expectedPreferences = { ...contactPreferencesViewModel }.preferences;
+                const currentValue = expectedPreferences.find(e => e.key === 'news').emailValue;
+                expectedPreferences.find(e => e.key === 'news').emailValue = !currentValue;
+
+                // Act
+                contactPreferencesModule.mutations[UPDATE_PREFERENCES](contactPreferencesViewModel, { preferences: expectedPreferences, preferenceVersionViewed: expectedPreferenceVersionViewed });
+                const actualPreferences = contactPreferencesViewModel.preferences;
+                const actualPreferenceVersionViewed = contactPreferencesViewModel.preferenceVersionViewed;
+
+                // Assert
+                expect(actualPreferences).toEqual(expectedPreferences);
+                expect(actualPreferenceVersionViewed).toEqual(expectedPreferenceVersionViewed);
+            });
+        });
+
+        describe(`${UPDATE_PREFERENCE} ::`, () => {
+            it('should update state with new preference value', () => {
+                // Arrange
+                const index = contactPreferencesViewModel.preferences.findIndex(e => e.key === 'news');
+                const expectedValue = !contactPreferencesViewModel.preferences[index].emailValue;
```

Comment:
What do you think of this?
```suggestion
                const expectedValue = !contactPreferencesViewModel.preferences.find(e => e.key === 'news')?.emailValue;
```

Reply:
But I need capture the `index` value so I can pass it in as a parameter a few lines below.

Reply:
Missed that 👍🏻